include/fizzy/fizzy.h

```diff
@@ -11,8 +11,86 @@
 extern "C" {
 #endif
 
+/// The opaque data type representing a module.
+struct FizzyModule;
+
+/// The opaque data type representing an instance (instantiated module).
+struct FizzyInstance;
+
+/// The data type representing numeric values.
+///
+/// i64 member is used to represent values of both i32 and i64 type.
+union FizzyValue
+{
+    uint64_t i64;
+    float f32;
+    double f64;
+};
+
+/// Result of execution of a function.
+typedef struct FizzyExecutionResult
+{
+    /// Whether execution ended with a trap.
+    bool trapped;
+    /// Whether function returned a value. Valid only if trapped == false
+    bool has_value;
+    /// Value returned from a function. Valid only if has_value == true
```

Comment:
```suggestion
    /// Value returned from a function. Valid only if has_value equals true.
```

Can this be true even if trap is true?

Reply:
> Can this be true even if trap is true?

@gumb0 ?

Reply:
Clarified in comment